sources/packages/backend/apps/api/src/route-controllers/cas-invoice/tests/e2e/cas-invoice.aest.controller.getInvoices.e2e-spec.ts

```diff
@@ -0,0 +1,273 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  createFakeCASInvoiceBatch,
+  createFakeDisbursementValue,
+  E2EDataSources,
+  saveFakeInvoiceFromDisbursementReceipt,
+  saveFakeInvoiceIntoBatchWithInvoiceDetails,
+} from "@sims/test-utils";
+import { SystemUsersService } from "@sims/services";
+import {
+  CASInvoiceStatus,
+  DisbursementValueType,
+  OfferingIntensity,
+  SupplierStatus,
+} from "@sims/sims-db";
+import { CASInvoiceAPIOutDTO } from "apps/api/src/route-controllers/cas-invoice/models/cas-invoice.dto";
+import { addDays } from "@sims/utilities";
+import { Between } from "typeorm";
+
+/**
+ * Use a period that will never be reached to delete all existing invoice batches
+ * and allow the retrieval of invoice batches to be tested.
+ */
+const CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE = new Date("2200-01-01");
+const CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE = new Date("2200-02-01");
+
+describe("CASInvoiceAESTController(e2e)-getInvoices", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let systemUsersService: SystemUsersService;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    systemUsersService = app.get(SystemUsersService);
+    db = createE2EDataSources(dataSource);
+  });
+
+  beforeEach(async () => {
+    // Delete existing CAS invoices and invoice details created between the CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE and the CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE dates.
+    const invoicesToDelete = await db.casInvoice.find({
+      select: { id: true },
+      where: {
+        invoiceStatusUpdatedOn: Between(
+          CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE,
+          CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE,
+        ),
+      },
+    });
+    for (const invoice of invoicesToDelete) {
+      await db.casInvoiceDetail.delete({
+        casInvoice: invoice,
+      });
+    }
+    await db.casInvoice.delete({
+      invoiceStatusUpdatedOn: Between(
+        CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE,
+        CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE,
+      ),
+    });
+    // Delete all existing invoice batches.
+    await db.casInvoiceBatch.delete({});
```

Comment:
Is there a reason to keep deleting the batches? I was under the impression we would do all deletions inside the period.